mmcv/fileio/file_client.py

```diff
@@ -410,6 +411,384 @@ def _list_dir_or_file(dir_path, list_dir, list_file, suffix,
+
+    def _map_path(self, filepath: Union[str, Path]) -> str:
+        """Map ``filepath`` to a string path whose prefix will be replaced by
+        :attr:`self.path_mapping`.
+
+        Args:
+            filepath (str): Path to be mapped.
+        """
+        filepath = str(filepath)
+        if self.path_mapping is not None:
+            for k, v in self.path_mapping.items():
+                filepath = filepath.replace(k, v)
+        return filepath
+
+    def _format_path(self, filepath: str) -> str:
+        """Convert a ``filepath`` to standard format of aws s3.
+
+        If the ``filepath`` is concatenated by ``os.path.join``, in a Windows
+        environment, the ``filepath`` will be the format of
+        's3://bucket_name\\image.jpg'. By invoking :meth:`_format_path`, the
+        above ``filepath`` will be converted to 's3://bucket_name/image.jpg'.
+
+        Args:
+            filepath (str): Path to be formatted.
+        """
+        return re.sub(r'\\+', '/', filepath)
+
+    def _parse_path(self, filepath: str) -> Tuple[str, str]:
+        """Parse bucket and object name from a given ``filepath``.
+
+        Args:
+            filepath (str or Path): Path to read data.
+
+        Returns:
+            bucket (str): Bucket name of aws s3.
+            obj_name (str): Object relative path to bucket.
+        """
+        filepath = self._map_path(filepath)
+        filepath = self._format_path(filepath)
+        parse_res = self.parse_bucket.match(filepath)
+        if not parse_res:
+            raise ValueError(
+                f"The input path '{filepath}' format is incorrect.")
+        bucket, obj_name = parse_res.groups()
+        return bucket, obj_name
+
+    def _check_bucket(self, bucket: str) -> bool:
+        """Check if bucket exists.
+
+        Args:
+            bucket (str): Bucket name
+
+        Returns:
+            bool: True if bucket is existing.
+        """
+        try:
+            self._client.head_bucket(Bucket=bucket)
+            return True
+        except self.check_exception:
+            return False
+
+    def _check_object(self, bucket: str, obj_name: str) -> bool:
+        """Check if object exists.
+
+        Args:
+            bucket (str): Bucket name
+            obj_name (str): Object name
+
+        Returns:
+            bool: True if object is existing.
+        """
+        try:
+            self._client.head_object(Bucket=bucket, Key=obj_name)
+            return True
+        except self.check_exception:
+            return False
+
+    def get(self, filepath: Union[str, Path]) -> memoryview:
+        """Read data from a given ``filepath`` with 'rb' mode.
+
+        Args:
+            filepath (str or Path): Path to read data.
+
+        Returns:
+            memoryview: A memory view of expected bytes object to avoid
+                copying. The memoryview object can be converted to bytes by
+                ``value_buf.tobytes()``.
```

Comment:
```suggestion
            memoryview: A memory view of expected bytes object to avoid
            copying. The memoryview object can be converted to bytes by
            ``value_buf.tobytes()``.
```